harness/libraries/apptest.py

```diff
@@ -547,14 +546,214 @@ def _start_test(self,
+            if not os.path.exists(f"./{test_id}/.influx_logged") and \
+                    not os.path.exists(f"./{test_id}/.influx_disabled"):
+                print(f"Attempting to log {test_id}")
+                if self._log_to_influx(test_id):
+                    print(f"Successfully logged {test_id}")
+                else:
+                    print(f"Unable to log {test_id}")
+
+        os.chdir(currentdir)
+
+    # Logs a single test ID to InfluxDB (when run AFTER a harness run, this class doesn't hold a single test ID)
+    def _log_to_influx(self, influx_test_id):
+        """ Check if metrics.txt exists, is proper format, and log to influxDB. """
+        currentdir = os.getcwd()
+        print("current directory in apptest:", currentdir)
+        runarchive_dir = f"{self.get_path_to_test()}/Run_Archive/{influx_test_id}"
+        os.chdir(runarchive_dir)
+        print("Starting influxDB logging in apptest:", os.getcwd())
+
+        if 'RGT_DISABLE_INFLUX' in os.environ and str(os.environ['RGT_DISABLE_INFLUX']) == '1':
+            print("InfluxDB logging is explicitly disabled with RGT_DISABLE_INFLUX=1")
+            print("Creating .influx_disabled file in Run_Archive")
+            print("If this was not intended, remove the .influx_disabled file and run the harness under mode 'influx_log'")
+            os.mknod('.influx_disabled')
+            os.chdir(currentdir)
+            return False
+        if not 'RGT_INFLUX_URI' in os.environ or not 'RGT_INFLUX_TOKEN' in os.environ:
+            print("RGT_INFLUX_URI and RGT_INFLUX_TOKEN required in environment to use InfluxDB")
+            os.chdir(currentdir)
+            return False
+
+        # Check if influx was disabled for this run
+        if os.path.exists('.influx_disabled'):
+            print("This harness test explicitly disabled influx logging. If this is by mistake, remove the .influx_disabled file and run again")
+            return False
+        # Check if the .influx_logged file already exists - it shouldn't, but just in case
+        if os.path.exists('.influx_logged'):
+            print("The .influx_logged file already exists.")
+            return False
+
+        import requests
+
+        influx_url = os.environ['RGT_INFLUX_URI']
+        influx_token = os.environ['RGT_INFLUX_TOKEN']
+
+        headers = {
+            'Authorization': "Token " + influx_token,
+            'Content-Type': "text/plain; charset=utf-8",
+            'Accept': "application/json"
+        }
+
+        # Inherited from environment or 'unknown'
+        # This may be set as `unknown` if run outside of harness job
+        influx_runtag = (
+            os.environ['RGT_SYSTEM_LOG_TAG']
+            if 'RGT_SYSTEM_LOG_TAG' in os.environ else 'unknown')
+        # Fields defined by subtest class
+        influx_app = self.getNameOfApplication()
+        influx_test = self.getNameOfSubtest()
+        # Machine name
+        if not 'LMOD_SYSTEM_NAME' in os.environ:
```

Comment:
This assumes the use of lmod, which may not always be the case. I believe there is a way to get the machine name from the harness itself.

Reply:
Could use `RGT_MACHINE_NAME` instead